src/main/java/org/cbioportal/web/PublicVirtualStudiesController.java

```diff
@@ -0,0 +1,207 @@
+package org.cbioportal.web;
+
+import com.mongodb.BasicDBObject;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import org.cbioportal.security.CancerStudyPermissionEvaluator;
+import org.cbioportal.security.VirtualStudyPermissionService;
+import org.cbioportal.service.CancerTypeService;
+import org.cbioportal.service.exception.CancerTypeNotFoundException;
+import org.cbioportal.service.util.SessionServiceRequestHandler;
+import org.cbioportal.utils.security.AccessLevel;
+import org.cbioportal.web.parameter.StudyViewFilter;
+import org.cbioportal.web.parameter.VirtualStudy;
+import org.cbioportal.web.parameter.VirtualStudyData;
+import org.cbioportal.web.parameter.VirtualStudySamples;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Controller
+@RequestMapping("/api/public_virtual_studies")
+public class PublicVirtualStudiesController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PublicVirtualStudiesController.class);
+    
+    @Value("${session.endpoint.publisher-api-key:}")
+    private String requiredPublisherApiKey;
+    
+    public static final String ALL_USERS = "*";
+    @Autowired
+    SessionServiceRequestHandler sessionServiceRequestHandler;
+    
+    @Value("${session.service.url:}")
+    private String sessionServiceURL;
+    
+    @Autowired
+    private CancerTypeService cancerTypeService;
+
+    @Autowired
+    private VirtualStudyPermissionService virtualStudyPermissionService;
```

Comment:
Can you update to Dep. Inj via Constructor

Reply:
done https://github.com/cBioPortal/cbioportal/pull/10829/commits/a5e6952fdf5d655b5f4a874d64abe0e6910d91be#diff-63e7ff33aeafd22afc2b9e835e7f1c54f039b5956daff471b3b095db7c090d33R55